src/sentry/workflow_engine/migration_helpers/alert_rule.py

```diff
@@ -1,20 +1,152 @@
+import logging
+
 from sentry.incidents.grouptype import MetricAlertFire
-from sentry.incidents.models.alert_rule import AlertRule
+from sentry.incidents.models.alert_rule import (
+    AlertRule,
+    AlertRuleThresholdType,
+    AlertRuleTrigger,
+    AlertRuleTriggerAction,
+)
+from sentry.integrations.services.integration import integration_service
 from sentry.snuba.models import QuerySubscription, SnubaQuery
 from sentry.users.services.user import RpcUser
 from sentry.workflow_engine.models import (
+    Action,
     AlertRuleDetector,
+    AlertRuleTriggerDataCondition,
     AlertRuleWorkflow,
+    DataCondition,
     DataConditionGroup,
+    DataConditionGroupAction,
     DataSource,
     Detector,
     DetectorState,
     DetectorWorkflow,
     Workflow,
     WorkflowDataConditionGroup,
 )
+from sentry.workflow_engine.models.data_condition import Condition
 from sentry.workflow_engine.types import DetectorPriorityLevel
 
+logger = logging.getLogger(__name__)
+
+
+def get_action_type(alert_rule_trigger_action: AlertRuleTriggerAction) -> str | None:
+    if alert_rule_trigger_action.sentry_app_id:
+        return Action.Type.SENTRY_APP
+
+    elif alert_rule_trigger_action.integration_id:
+        integration = integration_service.get_integration(
+            integration_id=alert_rule_trigger_action.integration_id
+        )
+        if not integration:
+            return None
+        for type in Action.Type.choices:
+            if type[0] == integration.provider:
+                return type[1]
+        return None
```

Comment:
nit: this could be something like
```suggestion
        try:
            return Action.Type(integration.provider)
        except Exception:
            return None
```